lib/emscripten/src/lib.rs

```diff
@@ -405,11 +405,18 @@ fn store_module_arguments(ctx: &mut Ctx, args: Vec<&str>) -> (u32, u32) {
     (argc as u32 - 1, argv_offset)
 }
 
-pub fn emscripten_set_up_memory(memory: &Memory, globals: &EmscriptenGlobalsData) {
+pub fn emscripten_set_up_memory(
+    memory: &Memory,
+    globals: &EmscriptenGlobalsData,
+) -> Result<(), String> {
     let dynamictop_ptr = globals.dynamictop_ptr;
     let dynamic_base = globals.dynamic_base;
 
+    if (dynamictop_ptr / 4) as usize >= memory.view::<u32>().len() {
+        return Err("dynamictop_ptr beyond memory len".to_string());
+    }
     memory.view::<u32>()[(dynamictop_ptr / 4) as usize].set(dynamic_base);
```

Comment:
I'm very tired, so I kind of missed this, but the bounds check here is something `WasmPtr` would have done for us and this is kind of new legacy code that we should probably just do right... I suppose I can replace it once this lands on master though